sparse_autoencoder/metrics/train/l0_norm_metric.py

```diff
@@ -0,0 +1,22 @@
+"""L0 (sparsity) norm metric."""
+from typing import final
+
+import torch
+
+from sparse_autoencoder.metrics.train.abstract_train_metric import (
+    AbstractTrainMetric,
+    TrainMetricData,
+)
+
+
+@final
+class L0NormMetric(AbstractTrainMetric):
+    """L0 (sparsity) norm metric."""
+
+    def calculate(self, data: TrainMetricData) -> dict[str, float]:
+        """Create a log item for Weights and Biases."""
+        # The L0 norm is the number of non-zero elements
+        # (We're averaging over the batch)
+        acts = data.learned_activations
+        value = (torch.sum(acts != 0) / acts.size(0)).item()
+        return {"l0_norm": value}
```

Comment:
A few suggestions here:

- The docstring should have an Args: and Returns: section
- The comment, that explains what L0 norm is, should be in the class docstring.
- Better to use descriptive names for variables (so rather than `acts` if you want to create a shorter reference, use `learned_activations`. 
- The resulting value to go to hugging face should have a descriptive name e.g. `learned_activations_mean_l0_norm`
- It's easier to understand what's going on if you split the value calc into it's underlying calcs, e.g.:

```
batch_size = data.learned_activations.size(0)
n_non_zero_activations = torch.count_nonzero(data.learned_activations)
average_l0_norm = n_non_zero_activations / batch_size
```